modules/ROOT/pages/deployment/storage/general-considerations.adoc

```diff
@@ -0,0 +1,173 @@
+====
+
+== Write Consistency
+
+Independent of the storage used and how many instances of the xref:{s-path}/storage-users.adoc[storage-users] service are running, Infinite Scale has mechanisms implemented to guarantee consistent writing to the backend for distributed setups. In any case, a file ID based on UUIDv4 is created to guarantee the uniqueness of files names. A proper file locking is implemented for writing metadata.
+
+== Local Storage
+
+Using local storage is a good choice if you want to test or plan to have a small productive instance. As drawback, scalability and failover is because of one monolithic block not given. Note that you can at anytime switch selected mount points defined to NFS which is described in the next section.
+
+To decide which filesystem for local storage can be used, see the  xref:prerequisites/prerequisites.adoc#supported-posix-compliant-filesystems[Supported POSIX-compliant Filesystems].
+
+We highly recommend, if the instance is planned to be used in production, to consider the follwing topics. For details and which environment variables necessary for the changes see the xref:deployment/general/general-info.adoc#default-paths[Default Paths] documentation:
+
+* To avoid filling up the filesystem and making the system unresponsive:
+** Separate via partitions the OS from where Infinite Scale metadata and blobs will be stored.
+
+* If you expect a data volume that can lead to a high demand in *search* and/or *thumbnail* space requirements - which would be stored without manual configuration in a subfolder using the `OCIS_BASE_DATA_PATH` as base path.
+** Though not required, the data path for services mentioned in xref:deployment/general/general-info.adoc#base-data-directory[Services That Can Deviate from the Base Data Path] can be deviated and individually configured. Using an additional partition/moint point is a possible way to achieve this.
+
+The following table describes where data should be stored when using Infinite Scale in production:
+
+{empty} +
+
+[role=center,width=70%,cols="40%,45%,60%",options="header"]
+|===
+| Envorinment Variable
+| Target
+| Space requirement
+
+| `OCIS_CONFIG_DIR`
+| Default (local)
+| Very low
+
+| `OCIS_BASE_DATA_PATH`
+| Partition
+| Medium
+|===
+
+== NFS
+
+Using NFS is benefical because the instance components are separated from the storage and can scale therefore independently. Such a setup is used when you plan a production environment with many users. As the same rules apply described in the topics of xref:local-storage[Local Storage], you mount instead using partitions NFS mount points. In addition, the Infinite Scale config is also stored on NFS using the xref:deployment/general/general-info.adoc#default-paths[OCIS_CONFIG_DIR] environment variable as this is a requirement for xref:deployment/container/orchestration/orchestration.adoc[orchestrated environments]. For details on NFS see the xref:deployment/storage/nfs.adoc[Network File System] documentation.
```

Comment:
```suggestion
Using NFS is beneficial because the instance components are separated from the storage and can scale independently. Such a setup is used when you plan a production environment with many users. The same rules apply as described in the xref:local-storage[Local Storage] section, but instead of using partitions you use NFS mount points. In addition, the Infinite Scale config is also stored on NFS using the xref:deployment/general/general-info.adoc#default-paths[OCIS_CONFIG_DIR] environment variable as this is a requirement for xref:deployment/container/orchestration/orchestration.adoc[orchestrated environments]. For details on NFS see the xref:deployment/storage/nfs.adoc[Network File System] documentation.
```